src/qibo/ui/plot_state.py

```diff
@@ -0,0 +1,200 @@
+    backend=None,
+    fig_width=16,
+    fig_height=8,
+):
+    """Plot the real and imaginary parts of the density matrix.
+
+    Given a :class:`qibo.models.circuit.Circuit`, plots the real and imaginary parts
+    of the final density matrix as separate 3D cityscape plots, side by side, and
+    with a gray ``z=0`` plane for the imaginary part.
+
+    Args:
+        circuit (:class:`qibo.models.circuit.Circuit`): Circuit to visualize.
+        title (str, optional): Title of the plot. Defaults to ``""``.
+        alpha (float, optional): Transparency level for the bars in the plot.
+            Defaults to :math:`0.5`.
+        colors (list, optional): A list of two colors for the positive and negative
+            parts of the density matrix. If ``None``, default colors will be used.
+            Defaults to ``None``.
+        backend (:class:`qibo.backends.abstract.Backend`, optional): backend
+            to be used in the execution. If ``None``, it uses
+            the current backend. Defaults to ``None``.
+        fig_width (int, optional): Width of the figure in inches. Defaults to ``16``.
+        fig_height (int, optional): Height of the figure in inches. Defaults to ``8``.
+
+    Returns:
+        tuple: Respectively, the figure, and axes for the real and the imaginary parts.
+    """
+
+    # if exec_circ has measurements then, error measure gates are present
+    if len(circuit.measurements) > 0:
+        raise ValueError(
+            "Circuit must not contain measurement gates for density matrix visualization"
+        )
+
+    backend = _check_backend(backend)
+    # Execute the circuit to get the state
+    exec_circ = backend.execute_circuit(circuit)
+
+    state = exec_circ.state()
+
+    # Create a density matrix from state vector
+    if not circuit.density_matrix:
+        state = backend.cast(state)
+        state = backend.np.outer(state, backend.np.conj(state))
+
+    nqubits = circuit.nqubits
+    row_names = _get_labels(nqubits)
+    column_names = _get_labels(nqubits)
+
+    matrix_real = state.real
+    matrix_imag = state.imag
+    del state
+
+    label_y, label_x = matrix_real.shape[:2]
+    xpos = np.arange(0, label_x, 1)
+    ypos = np.arange(0, label_y, 1)
+    xpos, ypos = np.meshgrid(xpos + 0.25, ypos + 0.25)
+    xpos = xpos.flatten()
+    ypos = ypos.flatten()
+    zpos = np.zeros(label_x * label_y)
+
+    dx = 0.5 * np.ones_like(zpos)
+    dy = dx.copy()
+    dzr = backend.np.reshape(matrix_real, (-1))
+    dzi = backend.np.reshape(matrix_imag, (-1))
+
+    fig = plt.figure(figsize=(fig_width, fig_height), facecolor="w")
+    ax1 = fig.add_subplot(1, 2, 1, projection="3d", computed_zorder=False)
+    ax2 = fig.add_subplot(1, 2, 2, projection="3d", computed_zorder=False)
+
+    max_dzr = np.max(dzr)
+    max_dzi = np.max(dzi)
+
+    max_plot_size = min(fig_width / 2.25, fig_height)
+    max_font_size = int(2 * max_plot_size)
+    max_zoom = 10 / (10 + np.sqrt(max_plot_size))
```

Comment:
I wonder why these choices of numbers and if this should / could be defined as `kwargs` of the function for users to adjust them if needed.